### PR TITLE
Feature/1896 allocate zero opp

### LINKED ISSUE #900: As a developer, I want to delete (if possible, otherwise deprecate) the UTIL_GlobalSchedulable, UTIL_GlobalSchedulable_TEST, and UTIL_IScheduled once all clients have the new master scheduled jobs running
These have been replaced by UTIL_MasterSchedulable, UTIL_MasterSchedulable_TEST, and UTIL_MasterSchedulableHelper.UTIL_IRecurring classes

### LINKED ISSUE #1896: Allocations: Donation of $0 allows allocation of arbitrarily large dollar amount.
Create a Donation
Navigate to Manage Allocations for that Opportunity.
Add any dollar amount to be allocated, e.g. $100.00
Save the record. 

This should not be allowed but it is. See screen shot.
screen shot 2016-01-14 at 10 40 04 am

Note: it is also possible to save legal values for percentage. This is weird but not actually wrong.

### PR DESCRIPTION
**Pull request by njjc**
# Warning
# Info
# Issues
Fixes #900
Fixes #1896

### CODE REVIEW

**Comment by njjc** [comment]
@dospromptman would you mind reviewing this? Thanks!

**Comment by dospromptman** [comment]
**lurch: review

**Comment by kbromer** [comment]
**lurch: review

**Comment by LurchTheButler** [comment]
#### General
- [ ] All new pages, classes or metadata conform to naming conventions

**Comment by kbromer** [comment]
**lurch: review apex visualforce

**Comment by LurchTheButler** [comment]
#### General
- [ ] All new pages, classes or metadata conform to naming conventions

#### Apex
- [ ] This request includes Apex
	- [ ] Sharing/Without/None has been explicitly determined and explained
	- [ ] FLS is appropriately checked and enforced
- [ ] Custom settings access is done through appropriate interfaces
- [ ] Methods are appropriately sized and scopes
- [ ] Inner classes utilized where appropriate
- [ ] Dependencies are done explicitly or have been noted where dynamic
- [ ] API Version is explicitly set to the latest or otherwise indicated as to why not
- [ ] Code is bulkified
	- [ ] No SOQL or DML in For loops
	- [ ] Async calls in For loops are properly rate limited
	- [ ] Code utilized DML wrapper where appropriate
- [ ] No new triggers have been added
- [ ] All currency operations are multi-currency capable and compliant
- [ ] Any new triggered functionality properly implements TDTM interface
	- [ ] A new TDTM record has been added to the install script for this class
	- [ ] Trigger recursion is handled appropriately
- [ ] All SOQL queries have been written as selectively as possible, and limited where appropriate
- [ ] Code is properly wrapped in try-catch and exceptions handled through the error handling framework
- [ ] This request includes batch or queueable apex
	- [ ] Batch apex is appropriately daisy-chained where required
- [ ] This request includes scheduable jobs
	- [ ] Scheduable jobs are properly injected into the scheduable framework

#### Tests
- [ ] Test coverage is > 85%
- [ ] Tests are in separate test classes
- [ ] Test conform to naming conventions
- [ ] All tests have positive and negative assertions
- [ ] Test data properly stands-up TDTM records where needed
- [ ] Tests make use of StartTest() and StopTest()
- [ ] Tests leverage different users and profiles

#### Visualforce
- [ ] This request includes VisualForce
	- [ ] Request meets mobile requirements
	- [ ] All pages are 508 compliant
	- [ ] All labels and text are translatable via Custom Labels
	- [ ] This VF has a controller
		- [ ] Controller sharing has been determined
		- [ ] Parameters are properly escaped
- [ ] CSRF protections are in place
- [ ] FLS is appropriately checked and enforced
- [ ] Business logic is not included in the controller
- [ ] Controller uses bind variables in SOQL
- [ ] Viewstate is managed through use of transients, etc.
	- [ ] Page includes custom CSS
	- [ ] Page include custom JS
		- [ ] Page includes new libraries or library requirements
		- [ ] Libraries are zipped and hosted in Static Resources
- [ ] Visualforce page utilizes existing VF components
- [ ] Templates are used where appropriate
- [ ] Visualforce page is an embedded component or page
- [ ] Visualforce page accepts user parameters for extendability
- [ ] User parameters are properly escaped and cleaned

**Comment by kbromer** [comment]
@dospromptman heroku went to sleep.  awake now.

File: src/package.xml [hunk 1]
```diff
@@ -112,7 +112,6 @@
         <members>OPP_OpportunityNaming</members>
         <members>OPP_OpportunityNamingBTN_CTRL</members>
         <members>OPP_OpportunityNaming_BATCH</members>
-        <members>OPP_OpportunityNaming_TDTM</members>
```
Comment (dospromptman) [inline src/package.xml]: `OPP_OpportunityNaming_TDTM` class is still referenced in `STG_InstallScript_TEST` and `TDTM_DefaultConfig`.

In `STG_InstallScript_TEST` it is used as part of a test record that is inserted into the TDTM table.  I assume this is okay, and the value used in this test row doesn't matter?

In `TDTM_DefaultConfig` this class is referenced as a deleted class.  Does this reference remain even though the class itself is being removed from the package?  At what point does that reference get removed, if ever?

Comment (dospromptman) [inline src/package.xml]: Dave says yes, so that these entries get removed from the TDTM table.

Comment (njjc) [inline src/package.xml]: Sorry I missed this, yes those are both by design.

**Pull request merged**